layerindex/tools/import_layers.py

```diff
@@ -0,0 +1,494 @@
+                sources = recipejs.get('sources', [])
+                import_child_items(recipe, Source, childlist=sources, key_fields=['url'])
+                patches = recipejs.get('patches', [])
+                import_child_items(recipe, Patch, childlist=patches, key_fields=['path'])
+                existing_deps = list(recipe.staticbuilddep_set.values_list('name', flat=True))
+                for dep in recipejs['staticbuilddeps']:
+                    depobj, created = StaticBuildDep.objects.get_or_create(name=dep)
+                    if created:
+                        depobj.save()
+                    elif dep in existing_deps:
+                        existing_deps.remove(dep)
+                    depobj.recipes.add(recipe)
+                for existing_dep in existing_deps:
+                    recipe.staticbuilddep_set.filter(name=existing_dep).first().recipes.remove(recipe)
+                package_configs = recipejs.get('package_configs', [])
+                import_child_items(recipe, PackageConfig, childlist=package_configs, custom_fields=['builddeps'], custom_field_cb=package_config_field_handler, key_fields=['feature'])
+
+                # RecipeFileDependency objects need to be handled specially (since they link to a separate LayerBranch)
+                existing_filedeps = list(recipe.recipefiledependency_set.values_list('id', flat=True))
+                filedeps = recipejs.get('filedeps', [])
+                for filedep in filedeps:
+                    target_layerbranch = layerbranch_idmap.get(filedep['layerbranch'], None)
+                    if target_layerbranch is None:
+                        #logger.debug('Skipping recipe file dependency on layerbranch %s, branch not imported' % filedep['layerbranch'])
+                        continue
+                    depobj, created = RecipeFileDependency.objects.get_or_create(recipe=recipe, layerbranch=target_layerbranch, path=filedep['path'])
+                    if created:
+                        depobj.save()
+                    elif depobj.id in existing_filedeps:
+                        existing_filedeps.remove(depobj.id)
+                for idv in existing_filedeps:
+                    RecipeFileDependency.objects.filter(id=idv).delete()
+
+            # Get list of layerbranches that currently exist, so we can delete any that
+            # we don't find in the remote layer index (assuming they are on branches
+            # that *do* exist in the remote index and are in the list specified by
+            # -b/--branch, if any)
+            existing_layerbranches = list(LayerBranch.objects.filter(branch__in=branch_idmap.values()).values_list('id', flat=True))
+
+            exclude_fields = ['id', 'layer', 'branch', 'yp_compatible_version', 'updated']
+            layercount = len(jsdata)
+            for i, layerbranchjs in enumerate(jsdata):
+                branch = branch_idmap.get(layerbranchjs['branch'], None)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are using the `get()` method to retrieve the value for a key from a dictionary. If the key is not present, then `get()` returns the value that is passed in its optional second parameter. If if the key is not present, then `None` is the default return value. Therefore, there is no need to explicitly specify `None` in the second parameter.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)

Similar issue at line numbers 317, 417, and 421.